### DISCUSSION TITLE
Cannot set string property from Command Line

### DISCUSSION THREAD

**Discussion by tomasmanu** [bonsai-rx/bonsai#1225] [Q&A]
Hi all!
 I'm trying to pass a path to the `Value` property of `String` to start the workflow from CLI but it always throws me this error: 
`Excepción no controlada: System.ArgumentNullException: El valor no puede ser nulo.
Nombre del parámetro: property
   en Bonsai.Configuration.PropertyAssignment.Parse(String property)
   en Bonsai.Program.<>c__DisplayClass20_0.<Main>b__13(String property)
   en Bonsai.Configuration.CommandLineParser.Parse(String[] args)
   en Bonsai.Program.Main(String[] args)`

The line I'm using is the following:
`bonsai for_batch_analysis.bonsai --start -p stringProperty="WIN_20230111_18_13_44_Pro (online-video-cutter.com).mp4"`

How can I set the string property correctly?

I leave attached the workflow with the video for analysis:
[using_command_line.zip](https://github.com/bonsai-rx/bonsai/files/10415303/using_command_line.zip)

**Comment by bruno-f-cruz** [comment]
Did you attach the wrong workflow by any chance? I don't see any property named `stringProperty` in the workflow.

**Reply by tomasmanu** [reply]
Yes sorry, I named it as `stringProperty` because I read the tutorial for Bonsai CLI and saw an example where the value of Int was passed as `intProperty`, so I use that sintaxis. But It may be wrong calling it like that. What I wanted to pass was the `Value` of `String`, which is its externalized property.

**Reply by bruno-f-cruz** [reply]
Does it work if you make that change then?

**Reply by tomasmanu** [reply]
Nope, it doesn't work either. I tried `Value` and `String` and I always got the same error:
`Excepción no controlada: System.ArgumentNullException: El valor no puede ser nulo.
Nombre del parámetro: property
   en Bonsai.Configuration.PropertyAssignment.Parse(String property)
   en Bonsai.Program.<>c__DisplayClass20_0.<Main>b__13(String property)
   en Bonsai.Configuration.CommandLineParser.Parse(String[] args)
   en Bonsai.Program.Main(String[] args)`

This is the `ExternalizedProperty` I want to change in the workflow:

![image](https://user-images.githubusercontent.com/69548554/212504655-011274c3-d81f-447a-8c18-f7b59f166ded.png)

What should I write in the command line?

**Reply by bruno-f-cruz** [reply]
`bonsai for_batch_analysis.bonsai --start -p "Value"="test"` seems to work on my machine.
As a side note, you should name your property something other than Value as that is one of the most common default property names.

**Reply by tomasmanu** [reply]
Hi Bruno, I tried it  but it didn't work either. However, I could make it work including `:` between `-p` and the property name, without any space, as Thomas Akam posted in the Google Group (https://groups.google.com/g/bonsai-users/c/cMQ1iIF_9NA/m/eavXE791BwAJ). Just like this:
 `bonsai for_batch_analysis.bonsai --start -p:"Value"="WIN_20230111_18_13_44_Pro (online-video-cutter.com).mp4"` 
I didn't know the two points and the lack of space between `-p` and the property name were necessary as they don't appear in the CLI doc (https://bonsai-rx.org/docs/articles/cli.html)

**Reply by bruno-f-cruz** [reply]
What bonsai version are you using? The parameter binding changed in >2.7

**Reply by tomasmanu** [reply]
Oh, that may be the problem... I'm using version 2.6.3. I haven't updated the version as I didn't want to cause errors in the workflows I'm using. I shall generate an environment for 2.6.3 version and download the newer one?
